examples/tests/timer_overflow/main.c

```diff
@@ -0,0 +1,71 @@
+#include "timer.h"
+#include "tock.h"
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+/*
+ * This test registers a "long running" timer and verifies that the
+ * timer fires at about the time when it's expected to.
+ *
+ * NOTE: Modify the "MAX_TICKS" macro in "libtock/alarm_timer.c" to
+ * test the implementation of timer_in without waiting for
+ * the alarm to overflow after 2^32 ticks (which is about 36 hours
+ * with a 32768Hz oscillator). Example: if you want to simulate the
+ * alarm overflowing after a second on a 32768Hz oscillator you can set
+ * "MAX_TICKS" to 32768. You can get the frequence of the alarm by calling
+ * alarm_internal_frequency.
+ */
+
+#define TIMER_LEN_MS 2000
+
+static bool fired = false;
+
+// helper function to get current time in milliseconds
+// from gettimeasticks
+static uint32_t get_time_ms(void) {
+  struct timeval tv;
+  gettimeasticks(&tv, NULL);
+  return (tv.tv_sec * 1000) + (tv.tv_usec / 1000);
+}
+
+// callback of type subscribe_upcall which will get called when timer fires
+static void timer_fire_callback(__attribute__ ((unused)) int unused0,
+                                __attribute__ ((unused)) int unused1,
+                                __attribute__ ((unused)) int unused2,
+                                void*                        user_data) {
+  fired = true;
+
+  uint32_t expected_timer_end_ms = *((uint32_t*)user_data);
+  uint32_t actual_fire_time_ms   = get_time_ms();
+  uint32_t difference_ms         = actual_fire_time_ms - expected_timer_end_ms;
+
+  printf("Timer Fired!\n");
+  printf("\tExpected timer to fire at %ld ms\n", expected_timer_end_ms);
+  printf("\tTimer actually fired at %ld ms\n", actual_fire_time_ms);
+  printf("\tDifference of %ld ms\n", difference_ms);
```

Comment:
I think this should somehow indicate test success.

Reply:
Addressed here: https://github.com/tock/libtock-c/pull/395/commits/39c4c5b30cee19c8a396ceff55df7b6e579de94f